lib/src/notification/notification.component.spec.ts

```diff
@@ -0,0 +1,249 @@
+        >
+    `,
+})
+class NotificationTestComponent {
+    className?: string;
+    dismissable?: boolean;
+    dismissTimeout?: number;
+    message?: string;
+    onClick() {}
+
+    onDismiss() {}
+    persist?: boolean;
+    position?: NotificationPosition;
+    style?: NotificationStyle;
+    theme?: NotificationTheme;
+    title?: string;
+    toggle: boolean;
+    actions?: Array<NotificationAction>;
+}
+
+describe("NotificationComponent", () => {
+    let component: NotificationTestComponent;
+    let fixture: ComponentFixture<NotificationTestComponent>;
+    let onDismiss: jasmine.Spy;
+    let onClick: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [CommonModule],
+            declarations: [NotificationComponent, NotificationTestComponent],
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(NotificationTestComponent);
+        component = fixture.componentInstance;
+
+        component.toggle = true;
+        onDismiss = spyOn(component, "onDismiss");
+        onClick = spyOn(component, "onClick");
+        component.message = "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.";
+        component.actions = [
+            { text: "Yes, I'm in", action: () => this.onToggle2(false) },
+            { text: "Ignore", action: () => this.onToggle2(false) },
+        ];
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    describe("Should render with defaults", () => {
+        it("slide-in style should render with bottom-left position and purple theme", () => {
+            expect(fixture.debugElement.query(By.css(".style-slide-in"))).toBeTruthy();
+            expect(fixture.debugElement.query(By.css(".bottom-left"))).toBeTruthy();
+            expect(fixture.debugElement.query(By.css(".theme-purple"))).toBeTruthy();
+        });
+
+        it("bar style should render with top position and purple theme", () => {
+            component.style = "bar";
+            fixture.detectChanges();
+
+            expect(fixture.debugElement.query(By.css(".style-bar"))).toBeTruthy();
+            expect(fixture.debugElement.query(By.css(".top"))).toBeTruthy();
+            expect(fixture.debugElement.query(By.css(".theme-purple"))).toBeTruthy();
+        });
+
+        it("Should dismiss when the default timer is done", async () => {
+            component.toggle = true;
+            component.dismissTimeout = 500;
+            component.onDismiss = () => (component.toggle = false);
+
+            fixture.detectChanges();
+
+            expect(fixture.debugElement.query(By.css(".open"))).toBeTruthy();
+            await fixture.whenStable();
+            await setTimeout(() => {
+                expect(fixture.debugElement.query(By.css(".open"))).toBeTruthy();
+            }, 600);
```

Comment:
instead of testing this asynchronously which might cause issues, you can mock the timers and synchronously travel through time by using fake times. You can read more about this [here](https://jasmine.github.io/2.0/introduction.html#section-Mocking_the_JavaScript_Timeout_Functions)

Reply:
The link above is a different scenario where the test object to be spied on is within timeout. It however reminded me of a more feasible solution [fakeAsync](https://angular.io/api/core/testing/fakeAsync)